File: Java/SelectionSort.java
// cruxiu :)
import java.util.ArrayList;
import java.util.Scanner;

public class SelectionSort {
    public static Integer [] selectionSort(Integer [] list){
      for (int i = 0; i < list.length - 1; i++)
      {
          int index = i;
          for (int j = i + 1; j < list.length; j++)
              if (list[j] < list[index])
                  index = j;

          int smallerNumber = list[index];
          list[index] = list[i];
          list[i] = smallerNumber;
      }
      return list;
    }
    public static void main(String a[]){
        Scanner scanner = new Scanner(System.in);
        ArrayList <Integer> arrayList = new ArrayList<Integer>();
        System.out.println("Please, enter the elements of the list.");
        while(scanner.hasNextInt()){
          arrayList.add(scanner.nextInt());
        }
        Integer [] array = arrayList.toArray(new Integer[0]);
        System.out.println("Now, we will sort the list. Your list now is:");
        for(int i = 0; i < array.length; i++){
           System.out.println(array[i]);
        }
        array = selectionSort(array);
        System.out.println("After sort, your list now is:");
        for(int i = 0; i < array.length; i++){
           System.out.println(array[i]);
        }
    }
}

File: Kotlin/SelectionSort.kt
// cruxiu :)
import java.util.ArrayList
import java.util.Scanner

class SelectionSort {
    fun selectionSort(list: Array<Integer>): Array<Integer> {
        for (i in 0 until list.size - 1) {
            var index = i
            for (j in i + 1 until list.size)
                if (list[j] < list[index])
                    index = j

            val smallerNumber = list[index].toInt()
            list[index] = list[i]
            list[i] = smallerNumber
        }
        return list
    }

    fun main(a: Array<String>) {
        val scanner = Scanner(System.`in`)
        val arrayList = ArrayList<Integer>()
        System.out.println("Please, enter the elements of the list.")
        while (scanner.hasNextInt()) {
            arrayList.add(scanner.nextInt())
        }
        var array = arrayList.toArray(arrayOfNulls<Integer>(0))
        System.out.println("Now, we will sort the list. Your list now is:")
        for (i in array.indices) {
            System.out.println(array[i])
        }
        array = selectionSort(array)
        System.out.println("After sort, your list now is:")
        for (i in array.indices) {
            System.out.println(array[i])
        }
    }
}
